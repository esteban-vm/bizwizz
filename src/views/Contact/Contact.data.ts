import type { InfoProps } from './Info'

const info: InfoProps[] = [
  {
    id: '1bcfb7c3-6138-445e-81ab-4109756943d0',
    detail: { subject: 'Address', content: 'Mountain View, California' },
    icon: 'location',
  },
  {
    id: '2ec98648-6ce9-4df3-9f36-178aa5a41814',
    detail: { subject: 'Email', content: 'email_us@bizwizz.com' },
    icon: 'envelop',
  },
  {
    id: '3e3c9837-aeb9-4ee9-92fe-eb524ba93f07',
    detail: { subject: 'Telephone', content: '650-210-8852' },
    icon: 'address-book',
  },
]

export default info
