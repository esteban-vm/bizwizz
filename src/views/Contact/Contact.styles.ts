import clsx from 'clsx'
import { commonStyles } from '@/styles'

const styles = {
  ...commonStyles,
  contact: clsx('flex'),
  column: clsx(
    'm-[0.8rem_1.2rem] rounded-md bg-lightgrey-2 p-[1.6rem_3rem] text-left text-[1.6rem] text-black',
    'first:flex first:basis-[30%] first:flex-col first:justify-evenly',
    'last:basis-[70%] [&>h3]:text-4xl [&>h3]:font-semibold [&>p]:first:mt-8'
  ),
}

export default styles
