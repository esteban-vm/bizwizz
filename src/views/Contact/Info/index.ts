export { default, type InfoProps } from './Info'
