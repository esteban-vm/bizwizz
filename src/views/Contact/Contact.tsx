import type { FC } from '@/types'
import { Wave } from '@/components'
import Info from './Info'
import info from './Contact.data'
import styles from './Contact.styles'

const Contact: FC = () => {
  return (
    <section aria-labelledby='contact_title' className={styles.wrapper('white')} id='contact'>
      <Wave is='black' />
      <div className={styles.header}>
        <h2 className={styles.heading} id='contact_title'>
          Contact Us
        </h2>
      </div>
      <div className={styles.contact}>
        <div className={styles.column}>
          <h3>Get in Touch</h3>
          <p>
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quibusdam debitis eos unde officiis tempore
            assumenda ab repellat facere dolore dignissimos a repudiandae iste fugiat, rerum ad voluptates veritatis
            tenetur laboriosam?
          </p>
          {info.map((item) => (
            <Info key={item.id} {...item} />
          ))}
        </div>
        <div className={styles.column}>&nbsp;</div>
      </div>
    </section>
  )
}

export default Contact
