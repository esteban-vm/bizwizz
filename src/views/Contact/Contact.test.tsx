import { render, cleanup, within, screen } from '@/tests'
import Contact from './Contact'
import info from './Contact.data'

describe('Contact:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<Contact />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const title = /contact us/i
    const subtitle = /get in touch/i
    const contactElem = screen.getByRole('region')
    expect(contactElem).toBeInTheDocument()
    expect(contactElem).toHaveAccessibleName(title)
    expect(contactElem).toHaveAttribute('id', 'contact')
    expect(within(contactElem).getByRole('presentation', { hidden: true })).toBeInTheDocument()
    expect(within(contactElem).getByRole('heading', { name: title, level: 2 })).toBeInTheDocument()
    expect(within(contactElem).getByRole('heading', { name: subtitle, level: 3 })).toBeInTheDocument()
    expect(within(contactElem).getByText(/lorem, ipsum/i)).toBeInTheDocument()
    expect(within(contactElem).getAllByRole('article')).toHaveLength(info.length)
    expect(within(contactElem).getByRole('form')).toBeInTheDocument()
  })
})
