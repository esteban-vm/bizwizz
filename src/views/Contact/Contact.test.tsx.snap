// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contact: > should render correctly 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="contact_title"
    class="section bg-white"
    id="contact"
  >
    <svg
      aria-hidden="true"
      class="h-40 w-full text-black"
      preserveAspectRatio="none"
      role="presentation"
      viewBox="0 0 500 150"
    >
      <path
        d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
        fill="currentColor"
        stroke="none"
      />
    </svg>
    <div
      class="section_header"
    >
      <h2
        class="section_heading"
        id="contact_title"
      >
        Contact Us
      </h2>
    </div>
    <div
      class="flex"
    >
      <div
        class="m-[0.8rem_1.2rem] rounded-md bg-lightgrey-2 p-[1.6rem_3rem] text-left text-[1.6rem] text-black first:flex first:basis-[45%] first:flex-col first:justify-evenly last:basis-[65%] [&>h3]:text-4xl [&>h3]:font-semibold [&>p]:first:mt-8"
      >
        <h3>
          Get in Touch
        </h3>
        <p>
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quibusdam debitis eos unde officiis tempore assumenda ab repellat facere dolore dignissimos a repudiandae iste fugiat, rerum ad voluptates veritatis tenetur laboriosam?
        </p>
        <article
          class="mt-8 flex items-center"
          id="1bcfb7c3-6138-445e-81ab-4109756943d0"
        >
          <div
            class="mr-[1.6rem]"
          >
            <svg
              class="h-[2.4rem] w-[2.4rem] fill-primary"
              role="img"
            >
              <use
                xlink:href="#icon-location"
              />
            </svg>
          </div>
          <div
            class="text-left"
          >
            <p
              class="text-[1.6rem] first:text-3xl first:font-semibold"
            >
              Address
            </p>
            <p
              class="text-[1.6rem] first:text-3xl first:font-semibold"
            >
              Mountain View, California
            </p>
          </div>
        </article>
        <article
          class="mt-8 flex items-center"
          id="2ec98648-6ce9-4df3-9f36-178aa5a41814"
        >
          <div
            class="mr-[1.6rem]"
          >
            <svg
              class="h-[2.4rem] w-[2.4rem] fill-primary"
              role="img"
            >
              <use
                xlink:href="#icon-envelop"
              />
            </svg>
          </div>
          <div
            class="text-left"
          >
            <p
              class="text-[1.6rem] first:text-3xl first:font-semibold"
            >
              Email
            </p>
            <p
              class="text-[1.6rem] first:text-3xl first:font-semibold"
            >
              email_us@bizwizz.com
            </p>
          </div>
        </article>
        <article
          class="mt-8 flex items-center"
          id="3e3c9837-aeb9-4ee9-92fe-eb524ba93f07"
        >
          <div
            class="mr-[1.6rem]"
          >
            <svg
              class="h-[2.4rem] w-[2.4rem] fill-primary"
              role="img"
            >
              <use
                xlink:href="#icon-address-book"
              />
            </svg>
          </div>
          <div
            class="text-left"
          >
            <p
              class="text-[1.6rem] first:text-3xl first:font-semibold"
            >
              Telephone
            </p>
            <p
              class="text-[1.6rem] first:text-3xl first:font-semibold"
            >
              650-210-8852
            </p>
          </div>
        </article>
      </div>
      <div
        class="m-[0.8rem_1.2rem] rounded-md bg-lightgrey-2 p-[1.6rem_3rem] text-left text-[1.6rem] text-black first:flex first:basis-[45%] first:flex-col first:justify-evenly last:basis-[65%] [&>h3]:text-4xl [&>h3]:font-semibold [&>p]:first:mt-8"
      >
        <h3
          id="contact_form_heading"
        >
          Shoot Us a Message
        </h3>
        <div>
          <form
            aria-labelledby="contact_form_heading"
            novalidate=""
          >
            <div
              class="my-6 flex justify-between"
            >
              <div
                class="basis-[48%]"
              >
                <input
                  autocorrect="off"
                  class="form-field"
                  id="name"
                  maxlength="30"
                  minlength="5"
                  name="name"
                  placeholder="Your name"
                  spellcheck="false"
                  type="text"
                />
              </div>
              <div
                class="basis-[48%]"
              >
                <input
                  class="form-field"
                  id="email"
                  name="email"
                  placeholder="Your email address"
                  type="email"
                />
              </div>
            </div>
            <div
              class="my-6 w-full"
            >
              <input
                class="form-field"
                id="subject"
                maxlength="100"
                minlength="5"
                name="subject"
                placeholder="Subject"
                type="text"
              />
            </div>
            <div
              class="my-6 w-full"
            >
              <textarea
                class="form-field h-80 resize-none"
                id="message"
                maxlength="500"
                minlength="10"
                name="message"
                placeholder="Your message"
              />
            </div>
            <button
              class="block min-w-[20rem] cursor-pointer select-none bg-gradient-120 from-transparent from-0% via-transparent via-50% to-50% bg-[length:250%] p-[1.6rem] text-2xl font-[500] uppercase transition-all duration-300 hover:translate-x-4 hover:bg-[100%] bg-primary to-white text-white hover:text-primary rounded-xl border border-solid border-primary text-center"
              type="submit"
            >
              Send message
            </button>
          </form>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
